<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>(13A) Exploring the Table Panel heat map | Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap</title>
  <meta name="description" content="updated version of nature protocol" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="(13A) Exploring the Table Panel heat map | Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="updated version of nature protocol" />
  <meta name="github-repo" content="baderlab/EnrichmentMap_Protocol" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="(13A) Exploring the Table Panel heat map | Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap" />
  
  <meta name="twitter:description" content="updated version of nature protocol" />
  

<meta name="author" content="Ruth Isserlin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="navigation-and-interpretation-of-the-enrichment-map.html"/>
<link rel="next" href="genemania.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#attributions"><i class="fa fa-check"></i>Attributions:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="materials.html"><a href="materials.html"><i class="fa fa-check"></i>Materials</a><ul>
<li class="chapter" data-level="" data-path="materials.html"><a href="materials.html#equipment"><i class="fa fa-check"></i>Equipment</a><ul>
<li class="chapter" data-level="" data-path="materials.html"><a href="materials.html#hardware"><i class="fa fa-check"></i>Hardware</a></li>
<li class="chapter" data-level="" data-path="materials.html"><a href="materials.html#software"><i class="fa fa-check"></i>Software</a></li>
<li class="chapter" data-level="" data-path="materials.html"><a href="materials.html#input-data"><i class="fa fa-check"></i>Input data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="software-installation.html"><a href="software-installation.html"><i class="fa fa-check"></i>(1-5) Software installation</a></li>
<li class="chapter" data-level="" data-path="pathway-enrichment-analysis.html"><a href="pathway-enrichment-analysis.html"><i class="fa fa-check"></i>Pathway enrichment analysis</a></li>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html"><i class="fa fa-check"></i>(6A) Gene list using g:Profiler</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html#specify-parameters"><i class="fa fa-check"></i>Specify parameters</a></li>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html#downlaod-gmt"><i class="fa fa-check"></i>Downlaod GMT</a></li>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html#run-query"><i class="fa fa-check"></i>Run Query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html#filter-enrichment-results"><i class="fa fa-check"></i>Filter enrichment results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-enrich.html"><a href="gprofiler-enrich.html#downlaod-enrichment-file"><i class="fa fa-check"></i>Downlaod Enrichment File</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-enrich.html"><a href="gsea-enrich.html"><i class="fa fa-check"></i>(6B) Ranked gene list using GSEA</a><ul>
<li class="chapter" data-level="" data-path="gsea-enrich.html"><a href="gsea-enrich.html#gsea_ts1"><i class="fa fa-check"></i>Loading the required data files into GSEA</a></li>
<li class="chapter" data-level="" data-path="gsea-enrich.html"><a href="gsea-enrich.html#specifying-parameters"><i class="fa fa-check"></i>Specifying parameters</a></li>
<li class="chapter" data-level="" data-path="gsea-enrich.html"><a href="gsea-enrich.html#running-gsea"><i class="fa fa-check"></i>Running GSEA</a></li>
<li class="chapter" data-level="" data-path="gsea-enrich.html"><a href="gsea-enrich.html#examining-gsea-results"><i class="fa fa-check"></i>Examining GSEA results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="visualization-of-enrichment-results-with-enrichmentmap.html"><a href="visualization-of-enrichment-results-with-enrichmentmap.html"><i class="fa fa-check"></i>(7-9) Visualization of enrichment results with EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html"><i class="fa fa-check"></i>(9A) g:Profiler - Creation of enrichment maps for results generated in Step 6A</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#parameter-specification"><i class="fa fa-check"></i>Parameter specification</a></li>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#specification-of-additional-optional-files"><i class="fa fa-check"></i>Specification of additional optional files:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#tuning-parameters"><i class="fa fa-check"></i>Tuning parameters</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-em.html"><a href="gsea-em.html"><i class="fa fa-check"></i>(9B) GSEA - Creation of enrichment maps for results generated in Step 6B</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#parameter-specification"><i class="fa fa-check"></i>Parameter specification</a></li>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#specification-of-additional-optional-files"><i class="fa fa-check"></i>Specification of additional optional files:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-em.html"><a href="gprofiler-em.html#tuning-parameters"><i class="fa fa-check"></i>Tuning parameters</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="navigation-and-interpretation-of-the-enrichment-map.html"><a href="navigation-and-interpretation-of-the-enrichment-map.html"><i class="fa fa-check"></i>(10-12) Navigation and interpretation of the enrichment map</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html"><i class="fa fa-check"></i>(13A) Exploring the Table Panel heat map</a><ul>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#expression-viewer"><i class="fa fa-check"></i>Expression Viewer</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#sorting-options"><i class="fa fa-check"></i>Sorting options</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#gene-options"><i class="fa fa-check"></i>Gene Options</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#expression-data-transformations"><i class="fa fa-check"></i>Expression data transformations</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#additional-functions"><i class="fa fa-check"></i>Additional Functions</a></li>
<li class="chapter" data-level="" data-path="a-exploring-the-table-panel-heat-map.html"><a href="a-exploring-the-table-panel-heat-map.html#export-results"><i class="fa fa-check"></i>Export Results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania.html"><a href="genemania.html"><i class="fa fa-check"></i>Search GeneMANIA</a><ul>
<li class="chapter" data-level="" data-path="genemania.html"><a href="genemania.html#show-connections-between-genes-in-gene-set-using-genemania"><i class="fa fa-check"></i>Show connections between genes in Gene set using GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="genemania.html"><a href="genemania.html#update-genemania-visual-style"><i class="fa fa-check"></i>Update GeneMANIA visual style</a></li>
<li class="chapter" data-level="" data-path="genemania.html"><a href="genemania.html#explore-genemania-network"><i class="fa fa-check"></i>Explore GeneMANIA network</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="pathwaycommons.html"><a href="pathwaycommons.html"><i class="fa fa-check"></i>Search Pathway commons</a></li>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html"><i class="fa fa-check"></i>Search String</a><ul>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#show-connections-between-genes-in-gene-set-using-string"><i class="fa fa-check"></i>Show connections between genes in Gene set using STRING</a></li>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#update-string-visual-style"><i class="fa fa-check"></i>Update STRING visual style</a></li>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#explore-string-network"><i class="fa fa-check"></i>Explore STRING network</a><ul>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#string-enrichments"><i class="fa fa-check"></i>STRING Enrichments</a></li>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#string-filtering"><i class="fa fa-check"></i>STRING filtering</a></li>
<li class="chapter" data-level="" data-path="string.html"><a href="string.html#string-selected"><i class="fa fa-check"></i>STRING selected</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="b-organization-and-clarification-of-the-network.html"><a href="b-organization-and-clarification-of-the-network.html"><i class="fa fa-check"></i>(13B) Organization and clarification of the network</a><ul>
<li class="chapter" data-level="" data-path="b-organization-and-clarification-of-the-network.html"><a href="b-organization-and-clarification-of-the-network.html#separate-two-different-phenotypes"><i class="fa fa-check"></i>Separate two different phenotypes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="annotate.html"><a href="annotate.html"><i class="fa fa-check"></i>(13C) Defining major biological themes</a><ul>
<li class="chapter" data-level="" data-path="annotate.html"><a href="annotate.html#annotate-the-network"><i class="fa fa-check"></i>Annotate the Network</a></li>
<li class="chapter" data-level="" data-path="annotate.html"><a href="annotate.html#manually-arrange-network"><i class="fa fa-check"></i>Manually arrange network</a></li>
<li class="chapter" data-level="" data-path="annotate.html"><a href="annotate.html#resulting-network"><i class="fa fa-check"></i>Resulting Network</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="d-creation-of-a-simplified-network-view.html"><a href="d-creation-of-a-simplified-network-view.html"><i class="fa fa-check"></i>(13D) Creation of a simplified network view</a><ul>
<li class="chapter" data-level="" data-path="d-creation-of-a-simplified-network-view.html"><a href="d-creation-of-a-simplified-network-view.html#collapse-network"><i class="fa fa-check"></i>Collapse Network</a></li>
<li class="chapter" data-level="" data-path="d-creation-of-a-simplified-network-view.html"><a href="d-creation-of-a-simplified-network-view.html#scale-network"><i class="fa fa-check"></i>Scale Network</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html"><a href="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html"><i class="fa fa-check"></i>(13E) Manual arrangement of network nodes and updating of theme labels</a><ul>
<li class="chapter" data-level="" data-path="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html"><a href="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html#create-sub-network"><i class="fa fa-check"></i>Create Sub-network</a></li>
<li class="chapter" data-level="" data-path="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html"><a href="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html#publication-ready"><i class="fa fa-check"></i>Publication ready</a></li>
<li class="chapter" data-level="" data-path="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html"><a href="e-manual-arrangement-of-network-nodes-and-updating-of-theme-labels.html#rename-themes"><i class="fa fa-check"></i>Rename themes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="f-creation-of-a-subnetwork-that-highlights-a-specific-theme-subset.html"><a href="f-creation-of-a-subnetwork-that-highlights-a-specific-theme-subset.html"><i class="fa fa-check"></i>(13F) Creation of a subnetwork that highlights a specific theme subset</a></li>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html"><i class="fa fa-check"></i>(14 - 20) Exporting figures, creating legends and saving work</a><ul>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html#export-image"><i class="fa fa-check"></i>Export image</a></li>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html#save-network"><i class="fa fa-check"></i>Save Network</a></li>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html#network-creation-parameters"><i class="fa fa-check"></i>Network creation parameters</a></li>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html#legends"><i class="fa fa-check"></i>Legends</a></li>
<li class="chapter" data-level="" data-path="exporting-figures-creating-legends-and-saving-work.html"><a href="exporting-figures-creating-legends-and-saving-work.html#save"><i class="fa fa-check"></i>Save</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a-exploring-the-table-panel-heat-map" class="section level1">
<h1>(13A) Exploring the Table Panel heat map</h1>
<div id="expression-viewer" class="section level2">
<h2>Expression Viewer</h2>
<p>i.When a gene expression matrix is provided as input to EnrichmentMap, we can study the expression pattern of the genes included in enriched pathways.</p>
<ul>
<li>Click on either an individual node -<br />

<p align="center">
<img src="images/heatmap/indiv_node.png" alt="single node" width="75%"  />
</p></li>
<li>or a group of nodes -
<p align="center">
<img src="images/heatmap/group_node.png" alt="multi node" width="75%"  />
</p>
to generate a gene expression heat map that will appear in the <em>Heat Map</em> tab of the <em>Table Panel</em>.</li>
<li>If the analysis is based on GSEA results and a rank file is supplied, the ‘leading edge’ genes will be highlighted in yellow for <strong>individual</strong> node selections.
<p align="center">
<img src="images/heatmap/leading_edge.png" alt="leading edge" width="75%"  />
</p></li>
<li>Several options for heat map visualization are available.</li>
</ul>
<div class="rmd-tip">
<p>
If no expression file is provided to EnrichmentMap as input, it will automatically create a dummy expression file in which any gene found in the enrichment file will be given a placeholder expression value of 0.25, and any gene found in a pathway but not found in the enrichment results file will be assigned a placeholder expression value of ‘NA’ (not applicable). Therefore, clicking on any node in the enrichment map will show the genes used for the analysis as well as genes in the pathway that are not part of the query set.
</p>
</div>
</div>
<div id="sorting-options" class="section level2">
<h2>Sorting options</h2>
<ol start="2" type="i">
<li>
Adjust the <em>Sort</em> option.
</li>
</ol>
<ul>
<li>Sorting options include:
<ul>
<li>hierarchical clustering,</li>
<li>ranks or</li>
<li>clicking on any of the column headers in heatmap will sort by that column</li>
</ul></li>
<li>To change the sorting option:</li>
<li>click on the <em>Sort</em> button visible in the top left corner of the heat map table.</li>
<li><p align="center">
<img src="images/heatmap/sorting_options.png" alt="sort" width="75%"  />
</p></li>
<li>By default, the heat map is sorted by ranks if a rank file is supplied.</li>
<li>In the absence of a rank file, no sort is applied.</li>
<li>Additional rank files can be uploaded for comparison through the Settings menu located at the top right corner in the Heat Map panel.
<ul>
<li><p align="center">
<img src="images/heatmap/add_rank.png" alt="add rank" width="75%"  />
</p></li>
</ul></li>
<li>Choosing which rank file to sort from can be done by clicking on the Sort button and selecting the rank file by name.</li>
<li>Click the arrow next to the currently sorted column to invert the sort order.
<ul>
<li><p align="center">
<img src="images/heatmap/sort_direction.png" alt="sort direction" width="75%"  />
</p></li>
</ul></li>
<li>Click any of the column names to sort the selected column.</li>
</ul>
<div class="rmd-troubleshooting">
<p>
<strong>User does not know which sort option to choose</strong>
</p>
<p>
In the case of multiple conditions or conditions with variable expression profiles (e.g., cancer patient samples), hierarchical clustering tends to generate a more informative visualization
</p>
</div>
</div>
<div id="gene-options" class="section level2">
<h2>Gene Options</h2>
<ol start="3" type="i">
<li>
Define <em>Genes</em> you wish to visualize in the heat map
</li>
</ol>
<ul>
<li>Data can be viewed for all genes contained in the selected nodes (union of nodes) or</li>
<li>just for the genes common to selected nodes (intersection of nodes). By default,</li>
<li>all genes are shown.
<ul>
<li><p align="center">
<img src="images/heatmap/define_genes.png" alt="gene definition" width="75%"  />
</p></li>
</ul></li>
</ul>
</div>
<div id="expression-data-transformations" class="section level2">
<h2>Expression data transformations</h2>
<ol start="4" type="i">
<li>
Change the <em>Expressions</em> value visualization depending on your data type
</li>
</ol>
<ul>
<li>Data can be viewed:
<ul>
<li>as they were loaded (Values),</li>
<li>as row-normalized, in which case the row mean is subtracted from each value and then divided by the row’s standard deviation (Row Norm),</li>
<li>or as log-transformed (Log).</li>
<li>By default, set to (Values).
<ul>
<li><p align="center">
<img src="images/heatmap/expression_types.png" alt="expression types" width="75%"  />
</p></li>
</ul></li>
</ul></li>
</ul>
<ol start="5" type="i">
<li>
<em>Compress</em> heat map columns
</li>
</ol>
<ul>
<li>By default, all expression values are visible as individual columns in the heat map for expression sets with &lt;50 samples.</li>
<li>It is possible to compress the data into a single column by selecting one of the aggregation methods:
<ul>
<li>Dataset - Median,</li>
<li>Dataset - Max (maximum) or</li>
<li>Dataset - Min (minimum) —listed under Compress.</li>
</ul></li>
<li>If a CLS file has been uploaded, the expression set can be compressed using one column per defined sample group using the Class option.
<ul>
<li>Class - Median,</li>
<li>Class - Max (maximum) or</li>
<li>Class - Min (minimum)</li>
<li>Dataset - Median,</li>
<li>Dataset - Max (maximum) or</li>
<li>Dataset - Min (minimum)</li>
</ul></li>
<li><p>If the expression matrix contains ≥50 samples, EnrichmentMap will automatically compress the values to their <em>Dataset median</em> value by default.</p></li>
<li><p align="center">
<img src="images/heatmap/compress.png" alt="compression types" width="75%"  />
</p></li>
</ul>
<ol start="6" type="i">
<li>
Check <em>Values</em> to show the expression numerical values in addition to the heat map color scale.
</li>
</ol>
<ul>
<li><p align="center">
<img src="images/heatmap/values.png" alt="values" width="50%"  />
</p></li>
</ul>
</div>
<div id="additional-functions" class="section level2">
<h2>Additional Functions</h2>
<ol start="7" type="i">
<li>
Perform additional fine-tuning of the heat map using the <em>Settings</em> panel, accessed by clicking on the menu icon
</li>
</ol>
<ul>
<li><p align="center">
<img src="images/heatmap/settings.png" alt="settings" width="75%"  />
</p></li>
<li>This includes functionality to:
<ul>
<li><a href="genemania.html#genemania">Show in GeneMANIA...</a> - Use the genes in the heatmap and search GeneMANIA<span class="citation">(Warde-Farley et al. <a href="#ref-genemania">2010</a>)</span>.</li>
<li><a href="string.html#string">Show in STRING ...</a> - Use the genes in the heatmap and search STRING.</li>
<li><a href="pathwaycommons.html#pathwaycommons">Show in Pathway Commons...</a> - Use the genes in the heatmap and search Pathway Commons.</li>
<li>Add Rankings - add new ranking files,</li>
<li>Export as TXT... - export the heat map data as a tab-delimited text file</li>
<li>Export as PDF... - export the heat map data as a PDF image,</li>
<li>change the distance metric for hierarchical clustering,</li>
<li>Auto-focus HeatMap - turn on the node table heat map auto focus.</li>
<li>Sync Data Sets with Control Panel</li>
<li>Display only selected Data Sets - when using multiple datasets there can be many expression files. By default the heat map will only show the expression files of the selected dataset but if you would like to see them all then you can change this setting.</li>
<li>Auto Select leading edge</li>
</ul></li>
</ul>
<p><br></p>
<p>Genes are sorted using the GSEA rank file, highlighting the leading edge in yellow. All genes contained in selected nodes are shown, expression values are row-normalized, no compression is applied and individual expression numerical values are not shown. Column headings are colored according to sample phenotype. Red color refers to the first phenotype (mesenchymal), and blue to the second phenotype (immunoreactive).</p>
<p align="center">
<img src="images/heatmap/heatmap.png" alt="heatmap" width="75%"  />
</p>
<div class="rmd-troubleshooting">
<p>
<strong>Heat map column names are not colored by dataset</strong>
</p>
<p>
If the heat map columns are not colored for a GSEA analysis, make sure the phenotype names specified in the EnrichmentMap input panel here:
</p>
<p align="center">
<img src="images/heatmap/match_classes.png" alt="match" width="75%"  />
</p>
<p>
match the class names specified in the class file <a href="./data/Supplementary_Table7_TCGA_OV_RNAseq_classes.cls">Supplementary_Table7_TCGA_OV_RNAseq_classes.cls</a>.
</p>
</div>
</div>
<div id="export-results" class="section level2">
<h2>Export Results</h2>
<ol start="8" type="i">
<li>
The heat map can be exported to a text file for further analysis: click on the Settings icon of the heat map and select Export as TXT.
</li>
</ol>
<p align="center">
<img src="images/heatmap/export.png" alt="export" width="75%"  />
</p>
<ol start="9" type="i">
<li>
If only an individual node is selected, a dialog will offer to export the Leading edge only for GSEA analysis. If selected, only the highlighted genes will be exported; otherwise, the entire set of genes is saved.
</li>
</ol>
<p align="center">
<img src="images/heatmap/export_options.png" alt="export" width="75%"  />
</p>
<div class="rmd-troubleshooting">
<p>
<strong>The option to save only leading-edge genes is not available</strong>
</p>
<ul>
<li>
Selection includes more than one node or dataset contains no leading-edge information (i.e., was not built from GSEA results
</li>
<li>
The leading edge is available only for GSEA analyses. The option will appear only if the enrichment map was built with GSEA results and a rank file was specified
</li>
</ul>
</div>
<ol start="10" type="i">
<li>
Specify the file name and location and click <em>Save</em>.
</li>
</ol>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-genemania">
<p>Warde-Farley, David, Sylva L Donaldson, Ovi Comes, Khalid Zuberi, Rashad Badrawi, Pauline Chao, Max Franz, et al. 2010. “The Genemania Prediction Server: Biological Network Integration for Gene Prioritization and Predicting Gene Function.” <em>Nucleic Acids Research</em> 38 (suppl_2). Oxford University Press: W214–W220.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="navigation-and-interpretation-of-the-enrichment-map.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genemania.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["EnrichmentMap_Protocol.epub"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"new_session": false
});
});
</script>

</body>

</html>
